src/bin/nydus-image/main.rs

```diff
@@ -1182,32 +1202,140 @@ impl Command {
+                }
+            };
+        }
+        info!("Chunkdict metadata is saved at: {:?}", db_url);
+
+        // Connecting database and Generating chunk dictionary by algorithm "exponential_smoothing"
         let db_strs: Vec<&str> = db_url.split("://").collect();
         if db_strs.len() != 2 || (!db_strs[1].starts_with('/') && !db_strs[1].starts_with(':')) {
             bail!("Invalid database URL: {}", db_url);
         }
+        let algorithm = String::from("exponential_smoothing");
+        let _source_bootstrap_paths: Vec<PathBuf> = matches
+            .get_many::<String>("SOURCE")
+            .map(|paths| paths.map(PathBuf::from).collect())
+            .unwrap();
+
+        let (chunkdict, noise_points): (Vec<ChunkdictChunkInfo>, Vec<String>);
 
         match db_strs[0] {
             "sqlite" => {
-                let mut deduplicate: Deduplicate<SqliteDatabase> =
-                    Deduplicate::<SqliteDatabase>::new(db_strs[1])?;
-                deduplicate.save_metadata(bootstrap_path, config)?
+                let mut algorithm: deduplicate::Algorithm<SqliteDatabase> =
+                    deduplicate::Algorithm::<SqliteDatabase>::new(algorithm, db_strs[1])?;
+                let result = algorithm.chunkdict_generate()?;
+                chunkdict = result.0;
+                noise_points = result.1;
             }
             _ => {
-                bail!("Unsupported database type: {}, please use a valid database URI, such as 'sqlite:///path/to/database.db'.", db_strs[0])
+                bail!("Unsupported database type: {}, please use a valid database URI, such as 'sqlite:///path/to/chunkdict.db'.", db_strs[0])
             }
         };
-        info!("Chunkdict metadata is saved at: {:?}", db_url);
+
+        // Output noise point in DBSCAN clustering algorithm
+        info!(
+            "The length of chunkdict is {}",
+            Vec::<ChunkdictChunkInfo>::len(&chunkdict)
+        );
+        info!("It is not recommended to use image deduplication");
+        for image_name in noise_points {
+            info!("{}", image_name);
+        }
+
+        // Dump chunkdict to bootstrap
+        let features = Features::try_from(
+            matches
+                .get_one::<String>("features")
+                .map(|s| s.as_str())
+                .unwrap_or_default(),
+        )?;
+        let chunkdict_bootstrap_path = Self::get_bootstrap_storage(matches)?;
+        let config =
+            Self::get_configuration(matches).context("failed to get configuration information")?;
+        config
+            .internal
+            .set_blob_accessible(matches.get_one::<String>("config").is_some());
+        build_ctx.configuration = config;
+        build_ctx.blob_storage = Some(chunkdict_bootstrap_path);
+        build_ctx.blob_features = BlobFeatures::CAP_TAR_TOC;
```

Comment:
`blob_features` should be from the base bootstrap.

Reply:
Ok.